### PR TITLE
Chore: Upgrade Go to 1.21.5

### PR DESCRIPTION
**Pull request by tolzhabayev**
**What is this feature?**

Upgrading Go for Release Builds to `1.21.5`

**Why do we need this feature?**

Regular maintenance

### CODE REVIEW

File: scripts/build/ci-build/Dockerfile [hunk 1]
```diff
@@ -108,7 +108,7 @@ RUN rm dockerize-linux-amd64-v${DOCKERIZE_VERSION}.tar.gz
 # Use old Debian (LTS into 2024) in order to ensure binary compatibility with older glibc's.
 FROM debian:buster-20220822
 
-ENV GOVERSION=1.21.3 \
+ENV GOVERSION=1.21.5 \
```
Comment (tolzhabayev) [inline scripts/build/ci-build/Dockerfile:111]: @kminehart is this `Dockerfile` still in use?

File: contribute/backend/upgrading-go-version.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Upgrading Go Version
+
+Notes on upgrading Go version.
+
+Example PR: https://github.com/grafana/grafana/pull/79329
+
+## The main areas that need to change during the upgrade are:
+- https://github.com/grafana/grafana/blob/d8ecea4ed93efb2e4d64a5ee24bc08f3805f413d/scripts/drone/variables.star#L6
```
Comment (papagian) [inline contribute/backend/upgrading-go-version.md:8]: I think I can include here the need to re-generate the `drone.yml` by running `make drone`

File: contribute/backend/upgrading-go-version.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+- https://github.com/grafana/grafana/blob/d8ecea4ed93efb2e4d64a5ee24bc08f3805f413d/Makefile#L264
+- https://github.com/grafana/grafana/blob/d8ecea4ed93efb2e4d64a5ee24bc08f3805f413d/Dockerfile#L6
+
+### Additional files to change
+- Take a look in `.github/workflows` folder for what `go` version is being used there in various workflows.
```
Comment (papagian) [inline contribute/backend/upgrading-go-version.md:13]: I wonder if there is a way to define a variable once and refer to this from the workflows.

**Review by papagian — APPROVED** [review]

**Review by DanCech — APPROVED** [review]

File: scripts/build/ci-build/Dockerfile (hunk 1 above)
Comment (kminehart) [reply scripts/build/ci-build/Dockerfile:111]: yea

**Comment by kminehart** [comment]
make sure to do the same in enterprise :)

**Review by kminehart — APPROVED** [review]

File: scripts/build/ci-build/Dockerfile (hunk 1 above)
Comment (tolzhabayev) [reply scripts/build/ci-build/Dockerfile]: Can I remove the whole `publish_ci_build_container_image_pipeline` and it's references then?

Like that https://github.com/grafana/grafana/pull/79329/commits/a7bbedecef90a1026f68256c624603b29697db25? Please double check 🙏

File: contribute/backend/upgrading-go-version.md (hunk 2 above)
Comment (tolzhabayev) [reply contribute/backend/upgrading-go-version.md:8]: Good point, added https://github.com/grafana/grafana/pull/79329/commits/0769094faf9464da0d49c9d0842f3fde6fd5f953

File: contribute/backend/upgrading-go-version.md (hunk 3 above)
Comment (tolzhabayev) [reply contribute/backend/upgrading-go-version.md:15]: I only added the workflows as courtesy to keep an eye on, they are not necessarily have to have the same version as the main grafana. At least not all of them. 

Sharing a variable or a config is what I would rather avoid, to let them be separate and not too coupled to each other.

File: scripts/build/ci-build/Dockerfile (hunk 1 above)
Comment (tolzhabayev) [reply scripts/build/ci-build/Dockerfile]: @kminehart

File: scripts/drone/utils/images.star [hunk 4]
```diff
@@ -8,9 +8,10 @@ load(
     "nodejs_version",
 )
 
+# "go" image can be switched back to golang:{}-alpine once this is resolved https://github.com/mattn/go-sqlite3/pull/1177#issuecomment-1849176090
 images = {
     "git": "alpine/git:2.40.1",
-    "go": "golang:{}-alpine".format(golang_version),
+    "go": "golang:{}-alpine3.18".format(golang_version),
```
Comment (tolzhabayev) [inline scripts/drone/utils/images.star:14]: trying another image to avoid https://github.com/mattn/go-sqlite3/pull/1177#issuecomment-1849176090

**Review by kminehart — APPROVED** [review]

**Comment by nicolaerosia** [comment]
will a new version be released ? We need this as it includes a security fix

**Pull request merged**